app/controllers/users_controller.rb

```diff
@@ -26,10 +27,30 @@ def create
       render 'new'
     end
   end
+
+  def new_with_invitation_token 
+    invitation = Invitation.where(token: params[:token]).first
```

Comment:
find_by method uses where().take in  its definition. where().take returns whatever the database returns first where as where().first returns the first record according to the order of the primary key. Since the token is unique and it either exists or not, find_by is better in this instance.